fdbserver/worker.actor.cpp

```diff
@@ -2601,6 +2659,14 @@ ACTOR Future<Void> workerServer(Reference<IClusterConnectionRecord> connRecord,
 					ASSERT(req.initialClusterVersion >= 0);
 					LocalLineage _;
 					getCurrentLineage()->modify(&RoleLineage::role) = ProcessClass::ClusterRole::Storage;
+
+					// need to cleanup stale disk files if it's not the first time to run storage server on this worker
+					if (storageServerStartedCount > 0) {
+						TraceEvent(SevInfo, "CleanupStaleDiskStores", interf.id()).log();
+						std::vector<DiskStore> stores = getDiskStores(folder);
+						cleanupDiskStoreFuture = cleanupDiskStores(dbInfo, stores, locality);
```

Comment:
This will cancel the existing actor if it was already started.  It would be best to have the actor already running and waiting on an AsyncTrigger, which you can then .trigger() here.  

The cleanup actor can take a while to run because it should be using incremental delete, which was written because deleting a large file all at once causes performance problems for a bit.

Reply:
Continuing, cancelling the cleanup actor would cancel the incremental delete but the file would not be discovered the second time through the actor execution, that's why I'm saying to not cancel it and instead just set a trigger so it runs again once it is done.

Reply:
use a new way to track the cleanup actor. please check if it can address your concern.